File: Backend/AssignmentManagementSystem.BusinessLogicLayer/Entities/Class.cs
using System;
using System.Collections.Generic;

namespace AssignmentManagementSystem.BusinessLogicLayer.Entities;

public partial class Class
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public string Code { get; set; } = null!;
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<Assignment> Assignments { get; set; } = new List<Assignment>();
    public virtual ICollection<StudentClass> StudentClasses { get; set; } = new List<StudentClass>();
    public virtual ICollection<Subject> Subjects { get; set; } = new List<Subject>();
}

File: Frontend/src/types/class.ts
export interface Class {
    id: string;
    name: string;
    code: string;
    createdAt?: string;
    updatedAt?: string;
}

export interface CreateClassData {
    name: string;
    code: string;
}

export interface UpdateClassData {
    name: string;
    code: string;
}